### PR TITLE
Here comes a way better implementation of playsound

### PR DESCRIPTION
**Pull request by lxgr-linux**
still needs to be tested on OSX

### CODE REVIEW

**Comment by lxgr-linux** [comment]
To build on OSX try:
`go build -ldflags "-s -w" -buildmode=c-shared -o ./libplaysound.osx.so`
@MaFeLP

**Comment by MaFeLP** [comment]
## Review checklist
- [x] Works on macOS
- [ ] Works on Linux
- [ ] Works on Windows

## Suggestions
- GitHub action that automatically builds the library on/and pushes to master, if the path `playsound` has been modified.

**Comment by lxgr-linux** [comment]
It would be nice if you could push the compile osx binary.

**Comment by lxgr-linux** [comment]
> GitHub action that automatically builds the library on/and pushes to master, if the path playsound has been modified.

Thats a good idea, I also found an action that makes the go setup for osx very easy: https://github.com/marketplace/actions/action-gobrew

Did you have to install any compilers like gcc or xcode on macOS?

**Comment by MaFeLP** [comment]
Does not work on my Linux machine, but also does not crash. Here is `pokete.log`:

```log
[2022-07-27 14:08:32,309][INFO]: === Startup Pokete Grey Edition v0.8.0 ===
[2022-07-27 14:08:32,310][INFO]: [General] 0 mods are loaded: ()
[2022-07-27 14:08:32,846][INFO]: [General] Startup took 0.556561s
[2022-07-27 14:08:32,858][INFO]: [Playsound] playing /mnt/Data/pokete/assets/music/03 Chibi Ninja.mp3
[2022-07-27 14:08:39,243][INFO]: [Fight] Started between Steini (<class '__main__.Figure'>) lvl. 7and Horny (<class 'pokete_classes.providers.NatureProvider'>) lvl. 4
[2022-07-27 14:08:39,246][INFO]: [Playsound] playing /mnt/Data/pokete/assets/music/xDeviruchi - Decisive Battle (Loop).mp3
[2022-07-27 14:08:45,903][INFO]: [Poke][Steini] Used Tackle: {'eff': 1, 'n_hp': 3}
[2022-07-27 14:08:47,910][INFO]: [Poke][Horny] Used Meat Skewer: {'eff': 1, 'n_hp': 3}
[2022-07-27 14:08:59,391][INFO]: [Poke][Steini] Used Tackle: {'eff': 1, 'n_hp': 3}
[2022-07-27 14:09:01,096][INFO]: [Poke][Horny] Used Tackle: {'eff': 1, 'n_hp': 0}
[2022-07-27 14:09:04,111][INFO]: [Poke][Steini] Used Tackle: {'eff': 1, 'n_hp': 2}
[2022-07-27 14:09:05,917][INFO]: [Poke][Horny] Used Tackle: {'eff': 1, 'n_hp': 1}
[2022-07-27 14:09:08,981][INFO]: [Poke][Steini] Used Tackle: {'eff': 1, 'n_hp': 3}
[2022-07-27 14:09:10,888][INFO]: [Poke][Horny] Used Tackle: {'eff': 1, 'n_hp': 2}
[2022-07-27 14:09:13,200][INFO]: [Poke][Steini] Used Tackle: {'eff': 1, 'n_hp': 2}
[2022-07-27 14:09:15,106][INFO]: [Poke][Horny] Used Tackle: {'eff': 1, 'n_hp': 2}
[2022-07-27 14:09:19,573][INFO]: [Poke][Steini] Used Tackle: {'eff': 1, 'n_hp': 2}
[2022-07-27 14:09:21,479][INFO]: [Poke][Horny] Used Tackle: {'eff': 1, 'n_hp': 2}
[2022-07-27 14:09:25,145][INFO]: [Poke][Steini] Used Tackle: {'eff': 1, 'n_hp': 2}
[2022-07-27 14:09:27,251][INFO]: [Poke][Horny] Used Meat Skewer: {'eff': 1, 'n_hp': 4}
[2022-07-27 14:09:29,961][INFO]: [Poke][Steini] Used Tackle: {'eff': 1, 'n_hp': 3}
[2022-07-27 14:09:30,469][INFO]: [Playsound] playing /mnt/Data/pokete/assets/music/xDeviruchi - Decisive Battle (End).mp3
[2022-07-27 14:09:31,466][INFO]: [Poke][Steini] Gained 2xp (curr:52)
[2022-07-27 14:09:32,671][INFO]: [Fight] Ended, Steini(player) won
[2022-07-27 14:09:32,677][INFO]: [Playsound] playing /mnt/Data/pokete/assets/music/03 Chibi Ninja.mp3
[2022-07-27 14:09:34,271][INFO]: [Fight] Started between Steini (<class '__main__.Figure'>) lvl. 7and Vogli (<class 'pokete_classes.providers.NatureProvider'>) lvl. 4
[2022-07-27 14:09:34,276][INFO]: [Playsound] playing /mnt/Data/pokete/assets/music/xDeviruchi - Decisive Battle (Loop).mp3
[2022-07-27 14:09:42,027][INFO]: [Poke][Vogli] Used Power Peck: {'eff': 0.5, 'n_hp': 1}
[2022-07-27 14:09:46,743][INFO]: [Poke][Vogli] Used Tackle: {'eff': 1, 'n_hp': 2}
[2022-07-27 14:09:47,606][INFO]: [Playsound] playing /mnt/Data/pokete/assets/music/xDeviruchi - Decisive Battle (End).mp3
[2022-07-27 14:09:49,606][INFO]: [Fight] Ended, ran away
[2022-07-27 14:09:49,613][INFO]: [Playsound] playing /mnt/Data/pokete/assets/music/03 Chibi Ninja.mp3
[2022-07-27 14:09:54,277][INFO]: [Playsound] playing /mnt/Data/pokete/assets/music/Map.mp3
[2022-07-27 14:09:57,890][INFO]: [Playsound] playing /mnt/Data/pokete/assets/music/03 Chibi Ninja.mp3
[2022-07-27 14:09:59,710][INFO]: [General] Saved
[2022-07-27 14:09:59,710][INFO]: [General] Exiting...
```

**Comment by MaFeLP** [comment]
> > GitHub action that automatically builds the library on/and pushes to master, if the path playsound has been modified.
> 
> Thats a good idea, I also found an action that makes the go setup for osx very easy: https://github.com/marketplace/actions/action-gobrew
> 
> Did you have to install any compilers like gcc or xcode on macOS?

Go, xcode and gcc are all installed by default: https://github.com/actions/virtual-environments/blob/main/images/macos/macos-12-Readme.md

**Comment by lxgr-linux** [comment]
And you don't get any panic calls?

**Comment by MaFeLP** [comment]
> And you don't get any panic calls?

No.

**Comment by lxgr-linux** [comment]
Are you running it in a virtual env?

**Comment by MaFeLP** [comment]
Ok. Turns out it was my wrongly configured audio system.....

File: playsound/__init__.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import ctypes
+import time
```
Comment (MaFeLP) [inline playsound/__init__.py:2]: Unneeded import

```suggestion
```

File: playsound/__init__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import logging
+import sys
+from pathlib import Path
+
+
+def playsound(file):
+    """plays an mp3 file
```
Comment (MaFeLP) [inline playsound/__init__.py:9]:
```suggestion
    """Plays a mp3 file
```

File: playsound/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+## Build
+
+Linux
+```
```
Comment (MaFeLP) [inline playsound/README.md:4]:
```suggestion
```shell
```

File: playsound/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+go build -ldflags "-s -w" -buildmode=c-shared -o ./libplaysound.so
+```
+
+Windows
+```
```
Comment (MaFeLP) [inline playsound/README.md:9]:
```suggestion
```shell
```

File: playsound/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+GOOS=windows CGO_ENABLED=1 CC=x86_64-w64-mingw32-cc go build -ldflags "-s -w" -buildmode=c-shared -o ./libplaysound.dll
+```
+
+OSX
+```
```
Comment (MaFeLP) [inline playsound/README.md:14]:
```suggestion
```shell
```

File: README.md [hunk 6]
```diff
@@ -40,18 +40,20 @@ NOTE: In that case you first have to create the `~/.cache/pokete/` folder.
 
 For OSX:
 ```shell
-# pip install scrap_engine playsound pyobjc
+# pip install scrap_engine
 $ git clone https://github.com/lxgr-linux/pokete.git
 $ ./pokete/pokete.py
 ```
 
 For Windows:
```
Comment (MaFeLP) [inline README.md:48]: Some windows antivirus flag the .dll as malicious:

```suggestion
For Windows:

Some windows antivirus may flag the `libplaysound.dll` as malicious. If pokete crashes, please make sure that the .dll exists and is **not** in quarantine!

```

**Review by MaFeLP — CHANGES_REQUESTED** [review]
- `assets/AppImageBuilder.yml`:
  - [ ] Remove line 43
  - [ ] Remove unneeded go libraries (`libplaysound.osx.so` and `libplaysound.dll`) from playsound folder to make size smaller.

**Review by MaFeLP — APPROVED** [review]
LGTM

**Pull request merged**